### PR TITLE
Add some improvements

### PR DESCRIPTION
**Pull request by carlosgalvez-tiendeo**
Add endpoint argument to decorators in order to allow to use it inside blueprints

Improve performance of _check_permission method

Automatically add deny rules when an allow decorator is added
in order to deny access for all roles that not are in the decorator

Fix tests

### CODE REVIEW

File: flask_rbac/__init__.py [hunk 1]
```diff
@@ -256,7 +257,7 @@ def a_view_func():
             roles = _user.get_roles()
         return self._check_permission(roles, method, endpoint)
 
-    def allow(self, roles, methods, with_children=True):
+    def allow(self, roles, methods, with_children=True, endpoint=None):
```
Comment (shonenada) [inline flask_rbac/__init__.py:260]: Please add some use cases of using `endpoint` (e.g. in `docs/quickstart.rst.inc`) , for helping others understand the usage of this argument.

File: flask_rbac/__init__.py [hunk 2]
```diff
@@ -422,6 +426,21 @@ def _setup_acl(self):
             else:
                 role = self._role_model.get_by_name(rn)
             self.acl.allow(role, method, resource, with_children)
+
+        if not self.use_white:
```
Comment (shonenada) [inline flask_rbac/__init__.py:430]: Also, please add some documents into `docs/`.

**Comment by carlosgalvez-tiendeo** [comment]
Hi,

there is a problem in travis with wsgiref and python3 dependency compatibility.
Did you consider deprecating python2?

Thx in advance

**Comment by shonenada** [comment]
LGTM, @carlosgalvez-tiendeo Thanks a lot!

**Comment by shonenada** [comment]
Now available with version 0.5.0

**Pull request merged**